### PR TITLE
1.86 cherrypicks

### PR DESCRIPTION
**Pull request by Duciwuci**
Well, this has a breakings: https://yake.cloud/release-notes/v1.86/

But the breakings needs to be fixed on our side. Therefore here are just the Cherries :)

### CODE REVIEW

**Comment by Duciwuci** [comment]
Part of https://github.com/stackitcloud/ske-base/issues/917

File: helmcharts/garden-kube-apiserver/templates/deployment-kube-apiserver.yaml [hunk 1]
```diff
@@ -122,6 +122,13 @@ spec:
         - --service-account-issuer=https://kubernetes.default.svc
         - --service-account-signing-key-file=/srv/kubernetes/service-account-key/service_account.key
         - --service-account-key-file=/srv/kubernetes/service-account-key/service_account.key
+        {{- if .Values.apiServer.featureGates }}
```
Comment (nschad) [inline helmcharts/garden-kube-apiserver/templates/deployment-kube-apiserver.yaml:125]: Can we check if this is not possible now via the upstream chart?

Comment (Duciwuci) [reply helmcharts/garden-kube-apiserver/templates/deployment-kube-apiserver.yaml:125]: Si Senor, lemme check

Comment (Duciwuci) [reply helmcharts/garden-kube-apiserver/templates/deployment-kube-apiserver.yaml:125]: Well I don't see anything which makes feature gates configurable. Not even in latest: https://github.com/YAKEcloud/yake/blob/release-v1.93/helmcharts/garden-kube-apiserver/templates/deployment-kube-apiserver.yaml

Comment (Duciwuci) [reply helmcharts/garden-kube-apiserver/templates/deployment-kube-apiserver.yaml:125]: Or am I blind?

Comment (nschad) [reply helmcharts/garden-kube-apiserver/templates/deployment-kube-apiserver.yaml:125]: correct! maybe we should contribute it

File: gardener/kube-apiserver.yaml [hunk 2]
```diff
@@ -18,11 +18,10 @@ spec:
...
   chart:
     spec:
-      chart: garden-kube-apiserver
-      version: 4.2.0
+      chart: ./helmcharts/garden-kube-apiserver
       sourceRef:
-        kind: HelmRepository
-        name: gardener-community-charts
+        kind: GitRepository
+        name: 23ke
```
Comment (Duciwuci) [inline gardener/kube-apiserver.yaml]: Shouldn't we move this permanently to yake?

Comment (Duciwuci) [reply gardener/kube-apiserver.yaml]: https://yake.cloud/release-notes/v1.86/

Yes we should

File: gardener/kube-apiserver.yaml [hunk 3]
```diff
@@ -18,11 +18,10 @@ spec:
...
   chart:
     spec:
-      chart: garden-kube-apiserver
-      version: 4.2.0
+      chart: ./helmcharts/garden-kube-apiserver
       sourceRef:
-        kind: HelmRepository
-        name: gardener-community-charts
+        kind: GitRepository
+        name: yake
```
Comment (Duciwuci) [inline gardener/kube-apiserver.yaml:24]: @timebertt I moved `23ke` to `yake`. 

According to new https://yake.cloud/release-notes/v1.86/

**Comment by MichaelEischer** [comment]
/lgtm

looks like it could work. But please test run it before merging

**Review by MichaelEischer — APPROVED** [review]
Let's keep to downgraded etcd chart for now.

**Pull request merged**